lib/src/http.dart

```diff
@@ -37,6 +44,20 @@ class _PubHttpClient extends http.BaseClient {
       : this._inner = inner == null ? new http.Client() : inner;
 
   Future<http.StreamedResponse> send(http.BaseRequest request) async {
+    if (_shouldAddMetadata(request)) {
+      request.headers["X-Pub-OS"] = Platform.operatingSystem;
```

Comment:
Sorry. I thought I was more explicit than I was – at least we need to make sure we document how it works for the person handling the server side...